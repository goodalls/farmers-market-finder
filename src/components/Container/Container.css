@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,700,800");
.container {
  grid-column: 6/13;
  grid-row: 1/3;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column; }

.container li {
  margin: 10px;
  border: 2px solid #9BC47e;
  border-radius: 10px; }

.container p {
  font-weight: 700;
  text-align: center;
  color: #B69C69;
  line-height: 1.2em;
  padding: 2.5px 0; }

.loading {
  background-image: url("./carrot-gif.gif");
  background-size: 100%;
  background-repeat: no-repeat;
  height: 50%;
  width: 75%;
  background-position: center; }
