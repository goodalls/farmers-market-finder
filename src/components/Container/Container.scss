@import '../../styles/colors.scss';

.container {
  grid-column: 6/13;
  grid-row: 1/3;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
}

.container li {
  margin: 10px;
  border: 2px solid $primary-green;
  border-radius: 10px;
}

.container p {
  font-weight: 700;
  text-align: center;
  color: $font;
  line-height: 1.2em;
  padding: 2.5px 0;
}

.loading {
  background-image: url('./carrot-gif.gif');
  background-size: 100%;
  background-repeat: no-repeat;
  height: 50%;
  width: 75%;
  background-position: center; 
}