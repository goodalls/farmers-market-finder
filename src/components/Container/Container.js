import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import * as actions from '../../actions/actions';
import * as api from '../../utilities/api';
import { connect } from 'react-redux';
import './Container.css';

export class Container extends Component {
  markets = () => {
    return this.props.markets.map((market, index) => {
      return (
        <li
          key={index}
          onClick={event => this.handleSingleMarket(event, market.id)}
        >
          <p>Distance: {market.distance}</p>
          <p>{market.marketname}</p>
        </li>
      );
    });
  };

  componentDidMount() {}

  handleSingleMarket = async (event, id) => {
    const fetch = await api.marketDetails(id);
    this.props.marketDetails(id, fetch.marketdetails);
  };

  loadingRenderCheck = () => {
    if (!this.props.markets.length) {
      return <div className="loading" />;
    } else {
      return (
        <div>
          <Link to={'/map'}>{'Map View'}</Link>
          <ol>{this.markets()}</ol>
        </div>
      );
    }
  };

  render = () => {
    return <div className="container">{this.loadingRenderCheck()}</div>;
  };
}

export const mapStateToProps = store => ({
  markets: store.markets
});

export const mapDispatchToProps = dispatch => ({
  marketDetails:(id, detail) => dispatch(actions.addDetails(id, detail))
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
