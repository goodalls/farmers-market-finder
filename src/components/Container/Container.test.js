import React from 'react';
import { Container, mapStateToProps, mapDispatchToProps } from './Container';
import { shallow } from 'enzyme';

describe('CONTAINER', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Container markets={[]} user={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('MSTP and MDTP', () => {
    it('should define props MARKETS', () => {
      const mockStore = {
        markets: [],
        user: { favorites: [] }
      };
      const expected = [];
      const mapped = mapStateToProps(mockStore);
      expect(mapped.markets).toEqual(expected);
    });

    it('should define props USER', () => {
      const mockStore = {
        markets: [],
        user: { favorites: [] }
      };
      const expected = [];
      const mapped = mapStateToProps(mockStore);
      expect(mapped.user).toEqual(expected);
    });

    it('should call dispatch on MARKETDETAILS', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.marketDetails();
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
