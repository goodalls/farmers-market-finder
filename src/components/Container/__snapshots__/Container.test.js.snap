// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTAINER should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    markets={Array []}
    user={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="loading"
>
        <p>
                Loading...
        </p>
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Loading...
</p>,
        "className": "loading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loading"
>
          <p>
                    Loading...
          </p>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Loading...
</p>,
          "className": "loading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
